### PR TITLE
"Your" issues/pull requests

### PR DESCRIPTION
**Pull request by jacobbearden**
implements #183

### CODE REVIEW

**Comment by jacobbearden** [comment]
As stated on #183:

> I've got this mostly done on my fork, but I'm a little lost on handling highlighting (.selected) as I can't recognize a consistent pattern it uses.

**Comment by DrewML** [comment]
Thanks for the contribution, @jacobbearden! Can you provide a bit more context about what you're trying to get done with `.selected`? Trying to understand what we're doing relating to highlighting.

File: extension/page-detect.js [hunk 1]
```diff
@@ -11,10 +11,14 @@ window.pageDetect = (() => {
 
 	const isRepoTree = () => isRepo() && /\/tree\//.test(getRepoPath());
 
+	const isIssueSearch = () => location.pathname.indexOf('/issues') === 0;
```
Comment (DrewML) [inline extension/page-detect.js]: Can you add unit tests for `isIssueSearch` and `isPrSearch` [here](https://github.com/sindresorhus/refined-github/blob/006eb4ba974dca63022b8487e55c088216f813ab/test/page-detect.js)? The mock for `location` we have in place should already handle `pathname`

File: extension/content.js [hunk 2]
```diff
@@ -282,11 +299,18 @@ $(document).on('click', event => {
 	$target.closest('.js-details-container').toggleClass('refined-github-minimized');
 });
 
+// Handle issue list ajax
+$(document).on('pjax:end', event => {
+	addYoursMenuItem();
```
Comment (DrewML) [inline extension/content.js]: Not a huge deal, but I think it would make sense to move the page checking logic in here (`isPrSearch` and `isIssueSearch`), and just make `addYoursMenuItem` responsible for the DOM manipulation. Would match the pattern we follow in most other places.

**Comment by jacobbearden** [comment]
Added the tests and fixed the issues. That was pretty messy :stuck_out_tongue:

So the issue with `.selected` is, for example, when you're on the main issue list, and type `is:open is:issue` in to the search box none of them are highlighted, as expected. When you append `author:your-username` it highlights the "Created" button. If you append `assignee:your-username` it will change the highlight to the "Assigned" button.

Basically I'm not sure how to handle adding `.selected` to the new "Yours" button.

File: extension/content.js (hunk 2 above)
Comment (paulmolluzzo) [inline extension/content.js]: I agree. There's also a (minor) performance impact in that this function runs on every page, not just those that are issue/PR searches.

Comment (jacobbearden) [inline extension/content.js]: Something like this good?
```javascript
function addYoursMenuItem(type) {
	const username = getUsername();
	const $menu = $('.subnav-links');

	$menu.append(`
		<a href="/${type}?q=is%3Aopen+is%3Aissue+user%3A${username}" class="subnav-item">Yours</a>
	`);
}
```
and called like so:
```javascript
if (pageDetect.isIssueSearch()) {
	addYoursMenuItem('issues');
}
if (pageDetect.isPRSearch()) {
	addYoursMenuItem('pulls');
}
```

Comment (paulmolluzzo) [inline extension/content.js]:
```js
if (pageDetect.isIssueSearch() || pageDetect.isPRSearch()) {
    addYoursMenuItem('pulls');
}
```

Also, you might want to add a test that the button doesn't exist before adding it. Some pages don't do a full refresh of the DOM and you could have the button appended multiple times. (It might be premature optimization, but we had to do this pattern elsewhere.)

Adding something like this will make sure we don't add a button if it already exists:

```js
if ($(`.subnav-item[href="/${type}?q=is%3Aopen+is%3Aissue+user%3A${username}"]`).length > 0) {
    return;
}
```

Comment (jacobbearden) [inline extension/content.js]: The link has to be different for pulls and issues, thats why theres seperate ifs

Comment (jacobbearden) [inline extension/content.js]: The link has to be different for pulls and issues, thats why theres seperate ifs

Comment (jacobbearden) [inline extension/content.js]: Is there a better way than passing in a variable to handle that?

Comment (paulmolluzzo) [inline extension/content.js]: Get the `type` from the URL?

Comment (jacobbearden) [inline extension/content.js]: @paulmolluzzo done!

**Comment by paulmolluzzo** [comment]
> Basically I'm not sure how to handle adding .selected to the new "Yours" button.

I think we could just match the URL to the pattern. `is%3Aissue+is%3Aopen+user%3A[username]` _in the absence of another `.selected` element_. The buttons provided by GH have some hierarchy to them if you apply the multiple filters overlapping. The hierarchy (AFAICT) is: "Assigned", then "Created", then "Mentioned". So if you have all three filters on, only "Assigned" is highlighted. It's possible to chain this new filter with the other ones, so we don't want to have two buttons selected at the same time (I don't think...).

In this scenario the logic is something like:

```js
if ($('.subnav-links.selected').length === 0 && /user\%3A[username]/.test(location.search)) {
  // apply .selected class
}
```

What do you think?

// @jacobbearden @DrewML @sindresorhus @hkdobrev @ben-eb (you opened the original issue)

**Comment by sindresorhus** [comment]
@paulmolluzzo I agree with your comments.

**Comment by sindresorhus** [comment]
@jacobbearden Can you fix? And resolve the merge conflict.

**Comment by jacobbearden** [comment]
Yup, I'll do that momentarily

**Comment by ben-eb** [comment]
Looks good to me, @paulmolluzzo.

File: extension/content.js [hunk 3]
```diff
@@ -78,6 +78,24 @@ function addTrendingMenuItem() {
 	`);
 }
 
+function addYoursMenuItem() {
+	const page = location.pathname.split('/')[1];
```
Comment (sindresorhus) [inline extension/content.js]:
```js
const [, page] = location.pathname.split('/');
```

File: extension/content.js [hunk 4]
```diff
@@ -282,6 +300,13 @@ $(document).on('click', event => {
 	$target.closest('.js-details-container').toggleClass('refined-github-minimized');
 });
 
+// Handle issue list ajax
```
Comment (sindresorhus) [inline extension/content.js]: handle

File: extension/page-detect.js (hunk 1 above)
Comment (sindresorhus) [inline extension/page-detect.js]:
```js
const isIssueSearch = () => location.pathname.startsWith('/issues')
```

Same with `isPRSearch`

File: readme.md [hunk 5]
```diff
@@ -35,6 +35,7 @@ Our hope is that GitHub will notice and implement some of these much needed impr
 - Removes the comment box toolbar
 - Removes tooltips
 - Copy canonical link to file when [the `y` hotkey](https://help.github.com/articles/getting-permanent-links-to-files/) is used
+- Add link to issues/pull requests on your repositories
```
Comment (sindresorhus) [inline readme.md]: The description is not very clear.

**Comment by sindresorhus** [comment]
When I go to https://github.com/issues and click "Yours", the button disappears.

**Comment by sindresorhus** [comment]
@jacobbearden Are you able to finish this up? :)

**Comment by sindresorhus** [comment]
Closing for lack of response. @jacobbearden Happy to reopen if you ever get to this ;)
